sql/numbertype_test.go

```diff
@@ -246,6 +246,20 @@ func TestNumberSQL_BooleanFromBoolean(t *testing.T) {
 	assert.Equal(t, "INT8(0)", val.String())
 }
 
+func TestNumberSQL_NumberFromString(t *testing.T) {
+	val, err := Int64.SQL(NewEmptyContext(), nil, "not a number")
+	require.NoError(t, err)
+	assert.Equal(t, "not a number", val.ToString())
+
+	val, err = Float64.SQL(NewEmptyContext(), nil, "also not a number")
+	require.NoError(t, err)
+	assert.Equal(t, "also not a number", val.ToString())
+
+	val, err = Float64.SQL(NewEmptyContext(), nil, "still not a number")
+	require.NoError(t, err)
+	assert.Equal(t, "still not a number", val.ToString())
```

Comment:
(minor) I don't think you need these lines, since they're testing the same thing as lines 254 to 256.